### PR TITLE
ac_Fuusen OK and documented

### PR DESCRIPTION
**Pull request by inspectredc**
Wasn't sure if I'd be able to do the assets as it's unclear what object rom offsets they're from. If there is a way to find out, or if it's already known I can add that on to this PR

### CODE REVIEW

File: src/overlays/actors/ovl_Fuusen/ac_fuusen.c [hunk 1]
```diff
@@ -22,34 +36,437 @@ ActorProfile Fuusen_Profile = {
     /* */ aFSN_actor_draw,
     /* */ NULL,
 };
-#endif
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Fuusen/ac_fuusen/aFSN_actor_ct.s")
+xyz_t Init_Size = { 0.01f, 0.01f, 0.01f };
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Fuusen/ac_fuusen.c:40]: These are in-function static

File: src/overlays/actors/ovl_Fuusen/ac_fuusen.c [hunk 2]
```diff
@@ -22,34 +36,437 @@ ActorProfile Fuusen_Profile = {
...
+}
+
+void aFSN_actor_dt(Actor* thisx, UNUSED Game_Play* game_play) {
+    Fuusen* this = THIS;
+
+    if (this->shouldLookUp) {
+        Balloon_look_up();
+        return;
+    }
+    Balloon_kill();
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Fuusen/ac_fuusen.c:98]:
```suggestion
    if (this->shouldLookUp) {
        Balloon_look_up();
     
    }
    else {
        Balloon_kill();
    } 
```

File: src/overlays/actors/ovl_Fuusen/ac_fuusen.c [hunk 3]
```diff
@@ -22,34 +36,437 @@ ActorProfile Fuusen_Profile = {
...
+    UNUSED UNK_TYPE1 pad[0x8];
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Fuusen/ac_fuusen/func_80AAD580_jp.s")
+    func_80AADF10_jp(this, game_play);
+    if (this->birthTimer == 0) {
+        Actor_position_moveF(&this->actor);
+    } else if (this->birthTimer > 0) {
+        this->birthTimer--;
+    }
+    if (D_80106E10_jp != 0) {
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Fuusen/ac_fuusen.c:381]: `fuusen_DEBUG_mode_flag`?

File: src/overlays/actors/ovl_Fuusen/ac_fuusen.c (hunk 1 above)
Comment (inspectredc) [reply src/overlays/actors/ovl_Fuusen/ac_fuusen.c:40]: would this be the case for all data? not sure what the general ruling is for this on the repo

Comment (Prakxo) [reply src/overlays/actors/ovl_Fuusen/ac_fuusen.c:40]: If it has $ qualifier in the name from the maps, it's in function static, and generally we add them as in function static

File: src/overlays/actors/ovl_Fuusen/ac_fuusen.c (hunk 3 above)
Comment (inspectredc) [reply src/overlays/actors/ovl_Fuusen/ac_fuusen.c:381]: not sure how to confirm this is the case, but will change to that

File: src/overlays/actors/ovl_Fuusen/ac_fuusen.c (hunk 1 above)
Comment (inspectredc) [reply src/overlays/actors/ovl_Fuusen/ac_fuusen.c:40]: ah great, will adjust accordingly then!

File: include/tables/object_table.h [hunk 4]
```diff
@@ -410,7 +410,7 @@
 /* 0x18D */ DEFINE_OBJECT(object_00F48000, OBJECT_397)
 /* 0x18E */ DEFINE_OBJECT(object_00FD6000, OBJECT_398)
 /* 0x18F */ DEFINE_OBJECT(object_00FD9000, OBJECT_399)
-/* 0x190 */ DEFINE_OBJECT(object_011F5000, OBJECT_400)
+/* 0x190 */ DEFINE_OBJECT(act_balloon, OBJECT_400)
```
Comment (emilybrooks) [inline include/tables/object_table.h:413]: OBJECT_400 should be renamed to OBJECT_ACT_BALLOON. you will also have to rename it in the actor profile

File: src/overlays/actors/ovl_Fuusen/ac_fuusen.c [hunk 5]
```diff
@@ -1,55 +1,464 @@
...
+            // FAKE
+            dummy_label:
+                for (i = 0; i < ARRAY_COUNT(senkou_check_data); i++) {
+                    xyz_t_move(&screenPos, &this->actor.world.pos);
+                    yRot = senkou_check_data[i].z + this->actor.world.rot.y;
+                    screenPos.x += sin_s(yRot) * 80.0f;
+                    screenPos.z += cos_s(yRot) * 80.0f;
+                    unitFG = mFI_GetUnitFG(screenPos);
+                    if ((unitFG != NULL) &&
+                        ((*unitFG == 0x804) || (*unitFG == 0x809) || (*unitFG == 0x811) || (*unitFG == 0x819) ||
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Fuusen/ac_fuusen.c:250]: These could use the defines from `m_name_table`, but it's a bit inconsistent throughout the repo at the moment, so dunno

**Review by Prakxo — APPROVED** [review]
lgtm, mentioned one thing though

File: src/overlays/actors/ovl_Fuusen/ac_fuusen.c (hunk 5 above)
Comment (inspectredc) [reply src/overlays/actors/ovl_Fuusen/ac_fuusen.c:250]: oh nice, will use what we currently have here so its better for search replacing

**Review by emilybrooks — APPROVED** [review]

**Pull request merged**